src/PropertyHandler/ScalarExporter.php

```diff
@@ -6,13 +6,18 @@
 
 use Crell\Serde\Attributes\Field;
 use Crell\Serde\Deserializer;
+use Crell\Serde\SerdeError;
 use Crell\Serde\Serializer;
 use Crell\Serde\TypeCategory;
+use function array_key_exists;
 
 class ScalarExporter implements Exporter, Importer
 {
     public function exportValue(Serializer $serializer, Field $field, mixed $value, mixed $runningValue): mixed
     {
+        if ($field->isNullable && $value === null) {
+            return $serializer->formatter->serializeNull($runningValue, $field, $value);
+        }
```

Comment:
If the field is not nullable, then PHP would not have allowed a null into the field in the first place.  That makes the first check redundant.

Reply:
Fair point, remove the first part